### ISSUE TITLE
📑 Feature Request: Extensions for updated test script

### ISSUE THREAD

**Issue by kurysauce** [ersilia-os/ersilia#1228]
### Describe your feature request.

After developing the updated `test.py` command for Ersilia #1203 , I believe having a detailed report for model contributors and users would be helpful in debugging and understanding the models in context. 

The testing report summary, along with including the metrics measured in the `test.py` command should also include the following (not an exhaustive list):
- Inuputs that were tested, inputs that raised specific errors 
- Software Dependencies:  List of all libraries and tools used, including versions
  - Espeically helpful if users want to run and download the models locally
  - Warnings if models don't work on certain operating systems (MacOS, Windows, etc...) 
- Performance Tests:
  - metrics such as speed, memory usage
- Output checks, If errors are raised or ouputs are invalid, describe them in context:
  - Missing Values
  - Outliers
  - Formatting Errors

**Comment by kurysauce** [comment]
- Support the method `check_consistent_output` with checking more `Output Shape` types (as mentioned on the[ Ersilia Model Template Book](https://ersilia.gitbook.io/ersilia-book/ersilia-model-hub/incorporate-models/model-template))
   - Current type check supported are `Single`,`List`. Needs support for `Flexible List`, `Matrix`, `Serializable Object`
   - From a meeting with @miquelduranfrigola , we need to incorporate models that support this output shape type

**Comment by miquelduranfrigola** [comment]
Thanks @kurysauce
@DhanshreeA to bring you up to speed, this is not something we need to tackle immediately but it is a nice feature request from @kurysauce . Let's leave it in out to-do list for now.

**Comment by kurysauce** [comment]
Update `check_consistent_output` and comparison between bash & Ersilia run in `run_bash` methods to allow comparisons for data structure containing different output types. Current method only supports comparisons between outputs that are all numerical values and all strings.

**Comment by miquelduranfrigola** [comment]
This makes sense. Thanks @kurysauce

**Comment by GemmaTuron** [comment]
Hi @DhanshreeA 

@Abellegese has been working on the test script right? so is this on hold still or undergoing? If so, which issues from the original list have been tackled? Thanks

**Comment by Abellegese** [comment]
Yes @GemmaTuron this addressed in depth in new test pipeline including the playground. 

Currently it does not give list of packages with version. Thats only it, if its important will update to do that.

**Comment by GemmaTuron** [comment]
Hi @Abellegese 

Thanks, please give more details and/or a link to the documentation before we can close this issue. I need more info to understand the pipeline.

**Comment by Abellegese** [comment]
Gi @GemmaTuron yes I have a doc for playground. For model tester,  I will upload it  and let you know.

**Comment by Abellegese** [comment]
So this issue also has been addressed in the following cases:

1) Pytest for five ersilia CLI and all the associated function level testing using partial mocking.
      - Issue: https://github.com/ersilia-os/ersilia/issues/1368 @DhanshreeA this needs to be closed.
      - PR: https://github.com/ersilia-os/ersilia/pull/1397
 2)  Playground test: to test model behaviour using custom rules
      - Issue: https://github.com/ersilia-os/ersilia/issues/1368 @DhanshreeA this needs to be closed.
      - PR: https://github.com/ersilia-os/ersilia/pull/1383
  3) Model Tester: To test and remote and local models
      - PR: https://github.com/ersilia-os/ersilia/pull/1416

**Comment by Abellegese** [comment]
Hi @GemmaTuron the docs for model tester has been uploaded. You may check it.

**Comment by miquelduranfrigola** [comment]
Thanks. @DhanshreeA, I believe this is good to be closed?

**Comment by GemmaTuron** [comment]
The Test Module should be updated according to #1484 
@Abellegese Can you specify in this issue what changes are being made and report when completed? Thanks!

**Comment by GemmaTuron** [comment]
I'll close this to focus the discussion on the open issues about the test module
